src/components/common/SmartPrice.tsx

```diff
@@ -0,0 +1,65 @@
+import { calculatePrice, formatPrice, invertPrice, safeTokenName } from '@gnosis.pm/dex-js'
+import React, { useMemo, useState } from 'react'
+import BN from 'bn.js'
+import { TokenDetails } from 'types'
+
+import { SwapPrice } from 'components/common/SwapPrice'
+import { getMarket } from 'utils'
+
+interface Fraction {
+  denominator: BN
+  numerator: BN
+}
+
+interface Props {
+  buyToken: TokenDetails
+  sellToken: TokenDetails
+  price: Fraction
+}
+
+export const SmartPrice: React.FC<Props> = ({ buyToken, sellToken, price: priceFraction }) => {
+  const [isPriceInverted, setIsPriceInverted] = useState(false)
+  const { baseToken, quoteToken } = getMarket({ sellToken, receiveToken: buyToken })
+
+  const [price, priceInverse] = useMemo((): string[] => {
+    const buyOrderPrice = calculatePrice({
+      numerator: { amount: priceFraction.numerator, decimals: buyToken.decimals },
+      denominator: { amount: priceFraction.denominator, decimals: sellToken.decimals },
+    })
+    const buyOrderPriceInverse = invertPrice(buyOrderPrice)
+    const sellTokenIsQuote = buyToken === quoteToken
+    let price, priceInverse
+    if (sellTokenIsQuote) {
+      price = buyOrderPrice
+      priceInverse = buyOrderPriceInverse
+    } else {
+      price = buyOrderPriceInverse
+      priceInverse = buyOrderPrice
+    }
+    return [formatPrice(price), formatPrice(priceInverse)]
+  }, [buyToken, sellToken, quoteToken, priceFraction])
+
+  let priceDisplayed: string, quoteDisplayed: TokenDetails, baseDisplayed: TokenDetails
+  if (isPriceInverted) {
+    priceDisplayed = priceInverse
+    quoteDisplayed = baseToken
+    baseDisplayed = quoteToken
+  } else {
+    priceDisplayed = price
+    quoteDisplayed = quoteToken
+    baseDisplayed = baseToken
+  }
+
+  return (
+    <span title={`${priceDisplayed} ${safeTokenName(quoteDisplayed)} per ${safeTokenName(baseDisplayed)}`}>
+      {priceDisplayed}
+      &nbsp;
+      <SwapPrice
+        baseToken={baseToken}
+        quoteToken={quoteToken}
+        isPriceInverted={isPriceInverted}
+        onSwapPrices={(): void => setIsPriceInverted(!isPriceInverted)}
```

Comment:
```suggestion
        onSwapPrices={(): void => setIsPriceInverted((isPriceInverted) => !isPriceInverted)}
```

Reply:
Why is this better? I mean, is it possible that `isPriceInverted` doesn't have the right value when `onSwapPrices` is called?

I guess, not in a normal case. Only way I can think off is when accumulate several calls to the function. In that case, is your option better? I mean, should we allow 2 executions that happen before a re-rendender?

Reply:
i just think it's safer. if you feel strongly opposed to the change then it probably doesn't matter

Reply:
nup, I'm ok with the change actually. I'm more curious than anything